### DISCUSSION TITLE
Timed out error after action on navigation site

### DISCUSSION THREAD

**Discussion by batra-21** [getgauge/taiko#1781] [General]
I am performing some steps in navigation site www.getnada.com(use for temporary emails).Here i am able to open this site from my application and adding my email successfully using Add Inbox+-->email + domain-->Click Add Now! Button.
But after this action, I am getting timeout error for any statement.My next statement is open email but i am getting timeout error.

**Comment by sriv** [comment]
Can you share a sample script to help debug this?

**Reply by batra-21** [reply]
* Open new tab "https://getnada.com"
* Click on "Add inboxe" button
* Add email to inbox "genapprover1"
* Click "Add now" button without wait

```
step("Open new tab <url>", async (url) => {
    await commonPage.OpenNewTab(url)
});
 async OpenNewTab(url) {
        await commons.openNewTabWithUrl(url);
      }
 async openNewTabWithUrl(url){
    await openTab(url)
  }
step("Click on <decision> button", async (decision) => {
  await commons.sleep(1000)
  await commonPage.clickToOptionButton(decision)
  //await commons.sleep(1000)
});
   async clickToOptionButton(option){
        await commons.clickElementByXpath(printf(CommonPageLocator.decision_button_xpath, option))        
      }
  this.decision_button_xpath = "//button[contains(text(),'%s')]";
step("Add email to inbox <email>",async (email) => {
    //await commons.sleep(1000)
    await raiseDefectPage.addInbox(email)   
});
 async addInbox(email)
    {
            await commons.enterValueToFieldByXpath(RaiseDefectPageLocator.email_txb, email);
      await dropDown().select("getnada.com")
            
    }
step("Click <decision> button without wait", async (decision) => {
  await commonPage.clickToOptionButton(decision)
});
```
After Add now button click, On performing any action getting timeout error.

**Reply by zabil** [reply]
I am afraid the code sample provided here does not have enough information to debug the problem and run the script. Please share project that replicates only this issue (not your entire project) as a github repo or a Taiko script.

**Reply by batra-21** [reply]
Thanks zabil, I got solution on this.

**Reply by zabil** [reply]
For the benefit of others who might run into this issue can you please explain how you solved it?

**Reply by batra-21** [reply]
Instead of normal click method 
async clickElementByXpath(xpath) {
    await waitFor(async() => (await $(xpath).exists()))
    await click($(xpath));
  }
I used below method:
 async clickElementByXpathSpecial(xpath) {
    await waitFor(async() => (await $(xpath).exists()))
    await click($(xpath),{waitForEvents:['loadEventFired']});
  }
and it worked.
